bot/exts/utils/unfurl_message_links.py

```diff
@@ -0,0 +1,35 @@
+import discord
+from discord.ext.commands import Cog
+
+from bot.bot import Bot
+from bot.constants import Channels, Guild
+from bot.utils.channel import is_staff_channel
+from bot.utils.messages import extract_message_links, make_message_link_embed
+
+
+class UnfurlMsgLinks(Cog):
+    """Unfurl message links and send embeds containing information about the linked message."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+    @Cog.listener()
+    async def on_message(self, message: discord.Message) -> None:
+        """Unfurl messages linked in the message if any and send the message link embed."""
+        if (
+            not message.guild
+            or message.guild.id != Guild.id
+            or message.channel.id == Channels.incidents
+            or not is_staff_channel(message.channel)
+        ):
```

Comment:
Add a comment here describing the intended behavior?

Unfurl links only if the message containing the link:
- isn't in dms
- is not within the python discord
- is not within the incidents channel (handled in the incidents cog)
- and the message is in a staff channel


Also missing from here is the ability to configure other channels (which isn't required but would be amazing if we could get a channel whitelist and add staff channels to it rather than need to edit the code each time)